examples/snap_service.js

```diff
@@ -0,0 +1,77 @@
+var page = require('webpage').create();
+var server = require('webserver').create();
+var system = require('system');
+var port;
+
+if (system.args.length !== 2) {
+    console.log('Usage: snap_service.js <some port>');
+    phantom.exit(1);
+} else {
+    port = system.args[1];
+    var listening = server.listen(port, function(request, response) {
+
+        console.log(JSON.stringify(request, null, 4));
+
+        var url = null,
+            format = 'png',
+            quality = -1;
+
+        if (request.query) {
+            if (request.query.url)
+                url = request.query.url;
+            if (request.query.format)
+                format = request.query.format;
+            if (request.query.quality)
+                quality = request.query.quality *1;
+        }
+
+        if (request.post) {
+            if (request.post.url)
+                url = request.post.url;
+            if (request.post.format)
+                format = request.post.format;
+            if (request.post.quality)
+                quality = request.post.quality *1;
+        }
+
+        console.log("URL:", url);
+        console.log("Format:", format);
+        console.log("Quality:", quality);
+
+        if (!url || url.length == 0) {
+            response.write("");
+            response.close();
+        } else {
+
+            var page = require('webpage').create();
+
+            page.viewportSize = {
+                width: 800,
+                height: 600
+            };
+
+            page.open(url, function start(status) {
+
+                // Buffer is an Uint8ClampedArray
+                var buffer = page.renderBuffer(format, quality);
+
+                response.statusCode = 200;
+                response.setEncoding('binary');
```

Comment:
Can we also make it so `response.setEncoding('binary')` is unnecessary when `response.write` is used with a byte array?